src/main/java/ui/UpdateProgressWindow.java

```diff
@@ -0,0 +1,151 @@
+package ui;
+
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import updater.LabeledDownloadProgressBar;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * UpdateProgressWindow is the class that will display update download progresses
```

Comment:
`The class that represents the Window showing the progress of the downloading of the new update`